[2025-05-20T23:56:03.628+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: etl_reddit_pipeline.extract_reddit manual__2025-05-20T23:56:02.675914+00:00 [queued]>
[2025-05-20T23:56:03.633+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: etl_reddit_pipeline.extract_reddit manual__2025-05-20T23:56:02.675914+00:00 [queued]>
[2025-05-20T23:56:03.633+0000] {taskinstance.py:1359} INFO - Starting attempt 1 of 1
[2025-05-20T23:56:03.639+0000] {taskinstance.py:1380} INFO - Executing <Task(PythonOperator): extract_reddit> on 2025-05-20 23:56:02.675914+00:00
[2025-05-20T23:56:03.642+0000] {standard_task_runner.py:57} INFO - Started process 51 to run task
[2025-05-20T23:56:03.644+0000] {standard_task_runner.py:84} INFO - Running: ['airflow', 'tasks', 'run', 'etl_reddit_pipeline', 'extract_reddit', 'manual__2025-05-20T23:56:02.675914+00:00', '--job-id', '5', '--raw', '--subdir', 'DAGS_FOLDER/reddit_dag.py', '--cfg-path', '/tmp/tmpp_wns02l']
[2025-05-20T23:56:03.644+0000] {standard_task_runner.py:85} INFO - Job 5: Subtask extract_reddit
[2025-05-20T23:56:03.667+0000] {task_command.py:415} INFO - Running <TaskInstance: etl_reddit_pipeline.extract_reddit manual__2025-05-20T23:56:02.675914+00:00 [running]> on host 0bdedd2b68b9
[2025-05-20T23:56:03.701+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Ibrahim' AIRFLOW_CTX_DAG_ID='etl_reddit_pipeline' AIRFLOW_CTX_TASK_ID='extract_reddit' AIRFLOW_CTX_EXECUTION_DATE='2025-05-20T23:56:02.675914+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-05-20T23:56:02.675914+00:00'
[2025-05-20T23:56:03.702+0000] {logging_mixin.py:151} INFO - Error connecting to Reddit API: name 'praw' is not defined
[2025-05-20T23:56:03.702+0000] {taskinstance.py:1935} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/operators/python.py", line 192, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/operators/python.py", line 209, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/pipelines/reddit_pipeline.py", line 12, in reddit_pipeline
    posts = extract_posts(instance, subreddit, time_filter, limit)
  File "/opt/airflow/etls/reddit_etl.py", line 13, in extract_posts
    subreddit = reddit_instance.subreddit(subreddit)
AttributeError: 'NoneType' object has no attribute 'subreddit'
[2025-05-20T23:56:03.707+0000] {taskinstance.py:1398} INFO - Marking task as FAILED. dag_id=etl_reddit_pipeline, task_id=extract_reddit, execution_date=20250520T235602, start_date=20250520T235603, end_date=20250520T235603
[2025-05-20T23:56:03.713+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 5 for task extract_reddit ('NoneType' object has no attribute 'subreddit'; 51)
[2025-05-20T23:56:03.742+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2025-05-20T23:56:03.752+0000] {taskinstance.py:2776} INFO - 0 downstream tasks scheduled from follow-on schedule check
